apps/frontend/src/libs/components/balance-wheel-chart/libs/helpers/get-dot-offset/get-dot-offset.helper.ts

```diff
@@ -0,0 +1,70 @@
+import { DEFAULT_GRAPHICS_OFFSET } from "~/libs/components/balance-wheel-chart/libs/constants/constants.js";
+import {
+	DotOffsetX,
+	DotOffsetY,
+} from "~/libs/components/balance-wheel-chart/libs/enums/enums.js";
+import { type ChartArea } from "~/libs/types/types.js";
+
+import { getCenters } from "../get-centers/get-centers.helper.js";
+
+const getDotOffset = (
+	dotPosition: { x: number; y: number },
+	chartArea: ChartArea,
+): { offsetX: number; offsetY: number } => {
+	const {
+		absoluteCenterX,
+		firstPartCenterX,
+		firstPartCenterY,
+		secondPartCenterX,
+		secondPartCenterY,
+	} = getCenters(chartArea);
+
+	return {
+		offsetX: ((): number => {
+			switch (true) {
+				case dotPosition.x > secondPartCenterX: {
+					return DotOffsetX.RIGHTMOST;
+				}
+
+				case dotPosition.y < firstPartCenterY &&
+					dotPosition.x > absoluteCenterX: {
+					return DotOffsetX.LEFT_UPPER;
+				}
+
+				case dotPosition.y < firstPartCenterY: {
+					return DotOffsetX.UPPER_CENTER;
+				}
+
+				case dotPosition.y > secondPartCenterY &&
+					dotPosition.x < absoluteCenterX: {
+					return DotOffsetX.LEFT_BOTTOM;
+				}
+
+				case dotPosition.x < firstPartCenterX: {
+					return DotOffsetX.LEFTMOST;
+				}
+
+				default: {
+					return DEFAULT_GRAPHICS_OFFSET;
+				}
+			}
+		})(),
```

Comment:
Would it be better to use just separate `if`'s for every statement or to use `else if`'s for every statement except first and last?

Reply:
I think it will be fine to have just if's because it will look more neat than else-if. It will look more neat because you will have less square brackets. It is relevant for our case because every if-clause leads to return but if there was no return but some variable modification you would use else-if. I hope that makes sense :)

Reply:
I see, it really makes sense in that case. Thank you!